Sources/CacheAdvance/CacheHeaderHandle.swift

```diff
@@ -74,11 +74,11 @@ final class CacheHeaderHandle {
         try handle.write(data: expectedHeader.data(for: .offsetInFileAtEndOfNewestMessage))
     }
 
-    /// Checks if the expected header version matches the persisted header version.
+    /// Throws if the expected header version does not match the persisted header version.
     ///
-    /// - Returns: `true` if this object's header version matches that of `fileHeader`; otherwise `false`.
-    func canOpenFile() throws -> Bool {
-        canOpenFile(with: try memoizedMetadata())
+    /// - Throws: `CacheAdvanceError.incompatibleHeader` if this object's header version does not match that of `fileHeader`. May also throw a file reading error if the file can not be read.
+    func openFile() throws {
```

Comment:
The name of this method makes me expect it will give me a file handle back. What about `checkFile()` in the spirit of this new similar Apple API ([source](https://github.com/DougGregor/swift-evolution/blob/fa58e0c4036ea3efbf4b264592aa99fb0f2f88f8/proposals/nnnn-structured-concurrency.md))?
![Screen Shot 2021-01-11 at 5 26 43 PM](https://user-images.githubusercontent.com/1791049/104257438-48b74380-5432-11eb-83e3-047751823e0b.png)